src/js/gutenberg-plugin.js

```diff
@@ -296,6 +186,195 @@ const ClassifAIGenerateTagsButton = () => {
+ * ClassifAI Text to Audio component.
+ */
+const ClassifAITSpeechSynthesisToggle = ( props ) => {
+	// State of the audio being previewed in PluginDocumentSettingPanel.
+	const [ isPreviewing, setIsPreviewing ] = useState( false );
+
+	const [ timestamp, setTimestamp ] = useState( new Date().getTime() );
+
+	// Indicates whether speech synthesis is supported for the current post.
+	let isFeatureSupported = false;
+
+	// Indicates whether speech synthesis is enabled for the current post.
+	let isSynthesizeSpeech = 'yes' === useSelect( select => select( 'core/editor' ).getEditedPostAttribute( 'classifai_synthesize_speech' ) )
+
+	// Post type of the current post.
+	let postType = useSelect( select => select('core/editor').getCurrentPostType() );
+
+	// Says if the post is currently being saved.
+	let isSavingPost = useSelect( select => select( 'core/editor' ).isSavingPost() );
+
+	// Says whether if speech synthesis is in progress.
+	let isProcessingAudio = useSelect( select => select( postAudioStore ).getIsProcessing() );
+
+	// Figure out if speech synthesis is supported by the current post.
+	if ( classifaiTextToSpeechData && classifaiTextToSpeechData.supportedPostTypes.includes( postType ) ) {
+		isFeatureSupported = true;
+	}
+
+	// The audio ID saved in the DB for the current post.
+	const defaultAudioId = useSelect( select => select('core/editor').getEditedPostAttribute( 'classifai_post_audio_id' ) );
+
+	// New audio ID in case it is regenerated manually or through publishing/updating the current post.
+	let audioId = props.audioId || defaultAudioId;
+
+	// Get the attachment data by audio ID.
+	let attachments = useSelect( select => select( 'core' ).getEntityRecords( 'postType', 'attachment', { include: [ audioId ] }) );
+
+	// Get URL for the attachment.
+	let sourceUrl = attachments && attachments.length > 0 && attachments[0].source_url;
+
+	const isProcessingAudioProgress = useRef( false );
+
+	// Handles playing/pausing post audio during preview.
+	useEffect( () => {
+		const audioEl = document.getElementById( 'classifai-audio-preview' );
+
+		if ( ! audioEl ) {
+			return;
+		}
+		if ( isPreviewing ) {
+			audioEl.play();
+		} else {
+			audioEl.pause();
+		}
+	}, [ isPreviewing ] );
+
+	useEffect( () => {
+		if ( isProcessingAudio ) {
+			isProcessingAudioProgress.current = true;
+		}
+
+		if ( isProcessingAudioProgress.current && ! isProcessingAudio ) {
+			setTimestamp( new Date().getTime() );
+		}
+	}, [ isProcessingAudio ] );
+
+	// Callback to refresh/regenrate audio manually.
+	function refreshAudio() {
+		if ( isSavingPost ) {
+			return;
+		}
+
+		wp.data.dispatch( 'core/editor' ).savePost();
+	}
+
+	// Fetches the latest audio file to avoid disk cache.
+	let cacheBustingUrl = `${ sourceUrl }?ver=${ timestamp }`;
+
+	return (
+		<>
+			<ToggleControl
+				label={ __( 'Generate audio for this post.' ) }
+				help={ isFeatureSupported
+					? __( 'ClassifAI will generate audio for the post when it is published or updated.' )
```

Comment:
@Sidsector9 did we leave these string untranslatable intentionally?

Reply:
@ravinderk  The PR is not up for review yet, it is marked as draft. Please do not pick it up for review until it is ready.